tests/SharedTests/TestLeads.cs

```diff
@@ -0,0 +1,620 @@
+        public void TestQualifyLeadRequestCreateContactSuccess()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+            lead.Id = orgAdminUIService.Create(lead);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference(),
+                CreateContact = true
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 1);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateOpportunityAccountSuccess()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+            lead.Id = orgAdminUIService.Create(lead);
+
+            var account = new Account();
+            account.Id = orgAdminUIService.Create(account);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference(),
+                CreateOpportunity = true,
+                OpportunityCustomerId = account.ToEntityReference()
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 1);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateOpportunityContactSuccess()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+            lead.Id = orgAdminUIService.Create(lead);
+
+            var contact = new Contact();
+            contact.Id = orgAdminUIService.Create(contact);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference(),
+                CreateOpportunity = true,
+                OpportunityCustomerId = contact.ToEntityReference()
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 1);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateAllSuccess()
```

Comment:
Missing assert that lead has been qualified and that a new Opportunity and Account have been created and that the opportunity points the the new account and existing contact